.github/workflows/_elixir.yml

```diff
@@ -232,114 +139,30 @@ jobs:
-        id: versions
-        uses: marocchino/tool-versions-action@v1.2.0
-      - uses: erlef/setup-beam@v1
-        id: setup-beam
-        with:
-          otp-version: ${{ steps.versions.outputs.erlang }}
-          elixir-version: ${{ steps.versions.outputs.elixir }}
-      - uses: actions/cache/restore@v4
-        name: Restore Elixir Deps Cache
-        id: cache
-        env:
-          cache-name: elixir-deps-${{ env.MIX_ENV }}
-        with:
-          path: |
-            elixir/deps
-            elixir/_build/${{ env.MIX_ENV }}
-          key: ubuntu-22.04-${{ runner.arch }}-${{ steps.setup-beam.outputs.elixir-version }}-${{ env.cache-name }}-${{ hashFiles('elixir/mix.lock') }}
-      - name: Install Dependencies
-        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
-        run: mix deps.get --only $MIX_ENV
-      - name: Compile Dependencies
-        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
-        run: mix deps.compile --skip-umbrella-children
-      - uses: actions/cache/save@v4
-        name: Save Elixir Deps Cache
-        if: ${{ github.ref_name == 'main' }}
-        env:
-          cache-name: elixir-deps-${{ env.MIX_ENV }}
+      - uses: ./.github/actions/setup-node
+      - uses: ./.github/actions/setup-elixir
         with:
-          path: |
-            elixir/deps
-            elixir/_build/${{ env.MIX_ENV }}
-          key: ubuntu-22.04-${{ runner.arch }}-${{ steps.setup-beam.outputs.elixir-version }}-${{ env.cache-name }}-${{ hashFiles('elixir/mix.lock') }}
+          mix_env: ${{ env.MIX_ENV }}
       - name: Compile
         run: mix compile --warnings-as-errors
-      - name: Download main branch DB dump
-        id: download-artifact
-        if: ${{ !contains(github.ref, env.MAIN_BRANCH) }}
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          set -xe
-
-          REPO="${{ github.repository }}"
-          WORKFLOW="cd.yml"
-          ARTIFACT_NAME="db-dump"
-          DESTINATION="apps/domain/priv/repo"
-
-          ARTIFACTS_URL=$(
-            gh api \
-              -H "Accept: application/vnd.github+json" \
-              -H "X-GitHub-Api-Version: 2022-11-28" \
-              "/repos/${REPO}/actions/workflows/${WORKFLOW}/runs?event=push&branch=${{ env.MAIN_BRANCH }}&status=success&per_page=1" \
-              --jq ".workflow_runs[0].artifacts_url"
-          )
-
-          DOWNLOAD_URL=$(
-            gh api \
-              -H "Accept: application/vnd.github+json" \
-              -H "X-GitHub-Api-Version: 2022-11-28" \
-              "${ARTIFACTS_URL}" \
-              --jq '.artifacts[] | select(.name == "'${ARTIFACT_NAME}'") | .archive_download_url'
-          )
-
-          set +x
-          curl -H "Accept: application/vnd.github+json" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -L -o "${DESTINATION}.zip" "$DOWNLOAD_URL"
-
-          set -x
-          unzip "${DESTINATION}.zip" -d "${DESTINATION}"
-          rm "${DESTINATION}.zip"
```

Comment:
This isn't directly related it is? I don't fully understand what we are doing here but if it is possible to decouple this from the setup, then I'd say it is better over in https://github.com/firezone/firezone/pull/4049.